StreetSmartAPI/StreetSmart.WPF.Tests/StreetSmartAPITests.cs

```diff
@@ -131,4 +132,133 @@ public void CloseViewer_Empty()
     Mock.Verify();
   }
 #endif
-}
+  [Fact]
+  public async Task AddOverlay_ValidJson_ReturnsOverlay()
+  {
+    var geoJsonOverlay = new GeoJsonOverlay("{\"type\":\"FeatureCollection\",\"features\":[]}", "myName", "mySrs", Color.AliceBlue);
+    _browserMock.Setup(x => x.IsDisposed).Returns(false);
+    _browserMock.Setup(x => x.GetBrowser()).Returns(_iBrowserMock.Object);
+    _browserMock.Setup(x => x.ExecuteScriptAsync(It.IsAny<string>())).Callback(() => _viewer.OnResult(new { id = 4 }, "AddOverlay1")).Verifiable(Times.Once);
+
+    var result = await _viewer.AddOverlay(geoJsonOverlay);
+
+    Assert.Equal(geoJsonOverlay, result);
+    Mock.Verify();
+  }
+
+  [Fact]
+  public async Task AddOverlay_InvalidJson_ThrowsStreetSmartJsonException()
+  {
+    var invalidJsonOverlay = new GeoJsonOverlay("{invalid json}", "myName", "mySrs", Color.AliceBlue);
+
+    await Assert.ThrowsAsync<StreetSmart.Common.Exceptions.StreetSmartJsonException>(() => _viewer.AddOverlay(invalidJsonOverlay));
+  }
+
+  [Fact]
+  public async Task AddOverlay_NullOverlay_ThrowsStreetSmartJsonException()
```

Comment:
I think we are missing GeoJson null test for full coverage of the "AddOverlay" method.